datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -432,7 +434,10 @@ public void run() {
     } finally {
       if (null != _consumer) {
         try {
+          _logger.error("Before consumer close");
           _consumer.close();
+          _logger.error("After consumer close and resetting failure flag");
+          _failure = false;
```

Comment:
Don't clear this flag here. `OnPartitionsRevoked` is called from another thread and it can result in commit().  Although the close does not complete before this method is called in new kafka versions, it is better to add fail-safe in our code, in case they change some behavior in the future.

Also, get rid of the debug logs.

Reply:
I am thinking not clearing this flag here would be taken care of when the thread starts again after close with _failure=false. T
Tested that the tests pass by removing this flag clearing.